test/unit/workDiv/src/WorkDivForKernelTest.cpp

```diff
@@ -0,0 +1,204 @@
+        // Depending on the GPU or the compiler the test below might fail because threadsPerBlock can be equal to
+        // threadsPerBlockLimit, which is the max device limit.
+        CHECK(threadsPerBlock < static_cast<Idx>(threadsPerBlockLimit));
+    }
+    else if constexpr(alpaka::accMatchesTags<Acc, alpaka::TagGpuHipRt>)
+    {
+        // Get threads per block from the workDiv found by examining the kernel function
+        auto const threadsPerBlock = workDiv.m_blockThreadExtent[0];
+        // Get hard limits
+        auto const kernelFunctionAttributes = alpaka::getFunctionAttributes<Acc>(bundeledKernel);
+        auto const threadsPerBlockLimit = kernelFunctionAttributes.maxThreadsPerBlock;
+        // Compare the workdiv value calculated using the kernel function to the device max limits for threads per
+        // block. The calculated value using kernel function is certainly less than or equal to the device max limits.
+        // For a common AMD GPU, expected values for threadsPerBlock and threadsPerBlockLimit are 1024 and 1024.
+        // Depending on the GPU or the compiler the test below might fail because threadsPerBlock can be less than
+        // threadsPerBlockLimit, which is the max device limit.
+        CHECK(threadsPerBlock == static_cast<Idx>(threadsPerBlockLimit));
+    }
+    else if constexpr(alpaka::accMatchesTags<
+                          Acc,
+                          alpaka::TagCpuSerial,
+                          alpaka::TagCpuThreads,
+                          alpaka::TagCpuOmp2Threads,
+                          alpaka::TagCpuTbbBlocks>)
+    {
+        // CPU must have only 1 thread per block. In other words, number of blocks is equal to number of threads.
+        CHECK(workDiv == WorkDiv{Vec{1024}, Vec{1}, Vec{1}});
+        // Test a new 1D workdiv. Threads per block can not be larger than 1 for CPU. Hence 2 is not valid.
+        auto const& workDiv1DUsingInitList = WorkDiv{Vec{1024}, Vec{2}, Vec{1}};
+        auto const isWorkDivValidForCpu
+            = alpaka::isValidWorkDivKernel<Acc>(dev, bundeledKernel, workDiv1DUsingInitList);
+        CHECK(isWorkDivValidForCpu == false);
+    }
+}
+
+using TestAccs2D = alpaka::test::EnabledAccs<alpaka::DimInt<2u>, std::uint32_t>;
+
+TEMPLATE_LIST_TEST_CASE("getFunctionAttributes.2D.withIdx", "[workDivKernel]", TestAccs2D)
+{
+    using Acc = TestType;
+    using Idx = alpaka::Idx<Acc>;
+    using Dim = alpaka::Dim<Acc>;
+    using Vec = alpaka::Vec<Dim, Idx>;
+    using WorkDiv = alpaka::WorkDivMembers<Dim, Idx>;
+    auto const platform = alpaka::Platform<Acc>{};
+    auto const dev = alpaka::getDevByIdx(platform, 0);
+
+    TestKernelWithManyRegisters kernel;
+    size_t val(200ul);
+    auto const& bundeledKernel = alpaka::KernelBundle(kernel, val);
+
+    // Test getValidWorkDivForKernel function for 1024x2 threads per grid.
+    auto const workDiv = alpaka::getValidWorkDivForKernel<Acc>(dev, bundeledKernel, Vec{1024, 2}, Vec{1, 1});
```

Comment:
please change all block, grid and thread sizes to `{Y,X}` in this case `{2,1024}. The indexing in alpaka is **NOT** `X,Y`